sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/FileSystemAsyncAPITest.groovy

```diff
@@ -0,0 +1,1261 @@
+        null     | null       | null        | null         | garbageLeaseID
+    }
+
+    def "List paths"() {
+        setup:
+        def dirName = generatePathName()
+        fscAsync.getDirectoryAsyncClient(dirName).create().block()
+
+        def fileName = generatePathName()
+        fscAsync.getFileAsyncClient(fileName).create().block()
+
+        when:
+        def response = fscAsync.listPaths().collectList().block().iterator()
+
+        then:
+        def dirPath = response.next()
+        dirPath.getName() == dirName
+        dirPath.getETag()
+        dirPath.getGroup()
+        dirPath.getLastModified()
+        dirPath.getOwner()
+        dirPath.getPermissions()
+//        dirPath.getContentLength() // known issue with service
+        dirPath.isDirectory()
+
+        response.hasNext()
+        def filePath = response.next()
+        filePath.getName() == fileName
+        filePath.getETag()
+        filePath.getGroup()
+        filePath.getLastModified()
+        filePath.getOwner()
+        filePath.getPermissions()
+//        filePath.getContentLength() // known issue with service
+        !filePath.isDirectory()
+
+        !response.hasNext()
+    }
+
+    def "List paths recursive"() {
+        setup:
+        def dirName = generatePathName()
+        fscAsync.getDirectoryAsyncClient(dirName).create().block()
+
+        def fileName = generatePathName()
+        fscAsync.getFileAsyncClient(fileName).create().block()
+
+        when:
+        def response = fscAsync.listPaths(new ListPathsOptions().setRecursive(true)).collectList().block().iterator()
+
+        then:
+        def dirPath = response.next()
+        response.hasNext()
+        def filePath = response.next()
+        !response.hasNext()
+    }
+
+    def "List paths return upn"() {
+        setup:
+        def dirName = generatePathName()
+        fscAsync.getDirectoryAsyncClient(dirName).create().block()
+
+        def fileName = generatePathName()
+        fscAsync.getFileAsyncClient(fileName).create().block()
+
+        when:
+        def response = fscAsync.listPaths(new ListPathsOptions().setUserPrincipalNameReturned(true)).collectList().block().iterator()
+
+        then:
+        def dirPath = response.next()
+        response.hasNext()
+        def filePath = response.next()
+        !response.hasNext()
+    }
+
+    def "List paths max results"() {
+        setup:
+        def dirName = generatePathName()
+        fscAsync.getDirectoryAsyncClient(dirName).create().block()
+
+        def fileName = generatePathName()
+        fscAsync.getFileAsyncClient(fileName).create().block()
+
+        when:
+        def response = fscAsync.listPaths(new ListPathsOptions().setMaxResults(1)).collectList().block().iterator()
+
+        then:
+        def dirPath = response.next()
+        response.hasNext()
+        def filePath = response.next()
+        !response.hasNext()
+    }
+
+    def "List paths max results by page"() {
+        setup:
```

Comment:
Based on the issue being addressed, I think the list paths by page methods are the only additions here that are really necessary because there can be different behavior in the sync and async clients. We don't otherwise maintain separate tests for sync and async. I'd rather move these few tests into the existing FileSystemApiTest file and delete the rest

Reply:
According to your comment, fixed in the new version.